simespresso/io_QE/tests/test_qe.py

```diff
@@ -0,0 +1,231 @@
+__author__ = 'jeremy'
+
+import os
+import unittest
+
+from simespresso.io_QE import espresso_class
+from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.cuds.particles import Particle, Particles
+
+
+class OutcomesTest(unittest.TestCase):
+
+    def setUp(self):
+        self.filename = 'pwtest.in'
+        with open(self.filename, 'w') as text_file:
+            text_file.write(_data_file_contents)
+
+    def test_espresso_data_file_read(self):
+        print('TEST OF READING QE INPUT FILE')
+        wrapper = espresso_class.qe_functions()
+        wrapper.read_espresso_input_file(self.filename)
+        expected_atom_positions = []
+        expected_atom_species = []
+        expected_atom_positions.append((1.0e-10, 2.0e-10, 3.0e-10))
+        expected_atom_positions.append((2.0e-10, 3.0e-10, 4.0e-10))
+        expected_atom_positions.append((3.0e-10, 4.0e-10, 5.0e-10))
+        expected_atom_positions.append((4.0e-10, 5.0e-10, 6.0e-10))
+        expected_atom_species.append('C')
+        expected_atom_species.append('C')
+        expected_atom_species.append('C')
+        expected_atom_species.append('C')
+        i = 0
+
+        for particle in wrapper.pc.iter_particles():
+            # print('expected:'+str(expected_atom_positions[i])+
+            # ' actual:'+str(particle))
+            # print('data:'+str(particle.data))
+            for expected_atom in expected_atom_positions:
+                # print('cur atom '+str(expected_atom)+
+                # ' trying to match '+str(particle.coordinates))
+                match = True
+                for dimension in range(0, 3):
+                    if(expected_atom[dimension] !=
+                            particle.coordinates[dimension]):
+                        match = False
+                        # print('no match:'+str(expected_atom[dimension])+
+                        # '!='+str(particle.coordinates[dimension]))
+                        break
+
+                if match is True:
+                    expected_atom_positions.remove(expected_atom)
+                    self.assertTrue(particle.data[CUBA.CHEMICAL_SPECIE]
+                                    == expected_atom_species[i])
+                    # print('cur atom matches')
+                    break
+
+    def test_espresso_data_file_write(self):
+        print('TEST OF WRITING QE INPUT FILE')
+        wrp = espresso_class.qe_functions()
+        espresso_input_filename = 'pw_generated.in'
+        wrp.SP = DataContainer()
+        wrp.pc = Particles('quantum_espresso_particles')
+        # write parameters for a particular working input file
+        SP = wrp.SP
+        pc = wrp.pc
+        SP[CUBA.TORQUE] = 'scf'
```

Comment:
This CUBA (and many more in this tests) are set but never used.  I suspect these need to be removed as you have already removed their usage in other places due to the comment https://github.com/simphony/simphony-espresso-dft/pull/7#discussion_r42113795.